tensorflow/python/keras/layers/convolutional.py

```diff
@@ -231,6 +231,33 @@ def build(self, input_shape):
     if tf_op_name == 'Conv1D':
       tf_op_name = 'conv1d'  # Backwards compat.
 
+    # Check if output shapes are valid
+    # They must not have 0 entries along any dimension
+    if self._channels_first:
+      for idx,dimension in enumerate(input_shape.as_list()[2:]):
```

Comment:
The code chunk under if and else are basically same, usually we discourage duplicated code. Could you create a small helper function to wrap the code? Thx!

Reply:
Sure, I'll add one